core/Config.php

```diff
@@ -231,13 +231,16 @@ public static function nest() {
 	}
 
 	/**
-	 * Change the active Config back to the Config instance the current active 
+	 * Change the active Config back to the Config instance the current active
 	 * Config object was copied from.
-	 * 
+	 *
 	 * @return Config Reference to new active Config instance
 	 */
 	public static function unnest() {
-		return self::set_instance(self::$instance->nestedFrom);
+		if (self::inst()->nestedFrom) {
+			self::set_instance(self::inst()->nestedFrom);
+		}
```

Comment:
Please use a warning here in an else so that we can catch these errors during dev.